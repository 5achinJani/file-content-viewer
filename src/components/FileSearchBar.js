import React from "react";

import {
  Button,
  Form,
  FormGroup,
  Label,
  Input,
  FormFeedback
} from "reactstrap";
import FontAwesome from "react-fontawesome";

export const FileSearchBar = ({ file, handleSubmit, handleChange }) => {
  const { fileName, error, isLoading } = file;
  let valid = null;
  if (error) {
    valid = false;
  }
  return (
    <React.Fragment>
      <Form name="searchForm" onSubmit={handleSubmit}>
        <FormGroup>
          <Label for="filename">Enter file name to browse its content:</Label>
          <Input
            required={true}
            type="text"
            name="filename"
            id="filename"
            placeholder="file name"
            value={fileName}
            onChange={handleChange}
            valid={valid}
          />
          <FormFeedback name="js-error-text">{error ? error : ""}</FormFeedback>
        </FormGroup>
        <FormGroup>
          <Button name="js-btn-submit" type="submit" disabled={isLoading}>
            View {isLoading ? <FontAwesome name="spinner" spin={true} /> : null}
          </Button>
        </FormGroup>
      </Form>
    </React.Fragment>
  );
};
