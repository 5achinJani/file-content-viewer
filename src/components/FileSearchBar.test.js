import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import { FileSearchBar } from "./FileSearchBar";

it("should render", () => {
  const props = {
    file: {
      fileName: "test",
      error: null,
      isLoading: false
    },
    handleSubmit: jest.fn(),
    handleChange: jest.fn()
  };
  const component = renderer.create(<FileSearchBar {...props} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("should not submit when loading", () => {
  const props = {
    file: {
      fileName: "test",
      error: null,
      isLoading: true
    },
    handleSubmit: jest.fn(),
    handleChange: jest.fn()
  };
  const component = shallow(<FileSearchBar {...props} />);
  const btnSubmit = component.find('[name="js-btn-submit"]');
  //const form = component.find('[name="searchForm"]');
  expect(btnSubmit.props().disabled).toBeTruthy();
  //form.simulate("submit");
  //expect(props.handleSubmit).not.toHaveBeenCalled();
});

it("should submit", () => {
  const props = {
    file: {
      fileName: "test",
      error: null,
      isLoading: false
    },
    handleSubmit: jest.fn(),
    handleChange: jest.fn()
  };
  const component = shallow(<FileSearchBar {...props} />);
  const btnSubmit = component.find('[name="js-btn-submit"]');
  const form = component.find('[name="searchForm"]');
  expect(btnSubmit.props().disabled).toBeFalsy();
  form.simulate("submit");
  expect(props.handleSubmit).toHaveBeenCalled();
});

it("should show loading component when loading", () => {
  const props = {
    file: {
      fileName: "test",
      error: null,
      isLoading: true
    },
    handleSubmit: jest.fn(),
    handleChange: jest.fn()
  };
  const component = shallow(<FileSearchBar {...props} />);
  const loader = component.find('[name="spinner"]');
  expect(loader.length).toBeTruthy();
});

it("should not show loading component when isLoading = false", () => {
  const props = {
    file: {
      fileName: "test",
      error: null,
      isLoading: false
    },
    handleSubmit: jest.fn(),
    handleChange: jest.fn()
  };
  const component = shallow(<FileSearchBar {...props} />);
  const loader = component.find('[name="spinner"]');
  expect(loader.length).toBeFalsy();
});

it("should render fileName and handle change event", () => {
  const fileName = "test";
  const props = {
    file: {
      fileName,
      error: null,
      isLoading: false
    },
    handleSubmit: jest.fn(),
    handleChange: jest.fn()
  };
  const component = shallow(<FileSearchBar {...props} />);
  const txtInput = component.find('[name="filename"]');
  expect(txtInput.props().value).toBe(fileName);
});

it("should show error", () => {
  const props = {
    file: {
      fileName: "test",
      error: "Fetch error",
      isLoading: false
    },
    handleSubmit: jest.fn(),
    handleChange: jest.fn()
  };
  const component = shallow(<FileSearchBar {...props} />);
  const elemError = component.find('[name="js-error-text"]');
  expect(elemError.contains(props.file.error)).toBeTruthy();
});
