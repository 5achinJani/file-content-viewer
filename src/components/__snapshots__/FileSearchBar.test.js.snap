// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<form
  className=""
  name="searchForm"
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <label
      className=" form-control-label"
      htmlFor="filename"
    >
      Enter file name to browse its content:
    </label>
    <input
      className="form-control"
      id="filename"
      name="filename"
      onChange={[Function]}
      placeholder="file name"
      required={true}
      type="text"
      value="test"
    />
    <div
      className="invalid-feedback"
      name="js-error-text"
    >
      
    </div>
  </div>
  <div
    className="form-group"
  >
    <button
      className="btn btn-secondary"
      disabled={false}
      name="js-btn-submit"
      onClick={[Function]}
      type="submit"
    >
      View 
    </button>
  </div>
</form>
`;
